xc/device.ml

```diff
@@ -724,6 +749,7 @@ let add (task: Xenops_task.t) ~xs ~devid ~netty ~mac ~carrier ?mtu ?(rate=None)
 	end;
 
 	Hotplug.wait_for_plug task ~xs device;
+	write_extra_xenserver_keys ~xs device extra_xenserver_keys;
```

Comment:
The reduce the scope for races, and to simplify the code, the new keys should all be written in the same transaction as the old ones. Rather than having a new function `write_extra_xenserver_keys`, it would be better to update `Generic.add_device` to create the new keys.